python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -274,6 +274,26 @@ def get_best_checkpoint(self,
         else:
             return min(checkpoint_paths, key=lambda x: x[1])[0]
 
+    def get_last_checkpoint(self, trial=None):
+        """Gets the last persistent checkpoint path of the provided trial,
+        according to the "training_iteration" metric.
+        Helper function that wraps Analysis.get_best_checkpoint().
+
+        Args:
+            trial (Trial): The log directory or an instance of a trial.
+            If None, load the latest trial automatically.
+
+        Returns:
+            Path for last checkpoint of trial
+        """
+        metric = "training_iteration"
+        mode = "max"
+
+        if trial is None:
+            trial = self.get_best_logdir(metric, mode)
```

Comment:
Hm, so if we run multiple trials, this will get the trial that ran for the most iterations, and will get the last checkpoint of that trial. 

Is this what we want? I thought we would want to get the last checkpoint of the _best_ trial.

In that case, shouldn't we just accept a metric and mode to fetch the best trial by that, and then return the last checkpoint of that trial?

Reply:
Hm, good question. I'm not sure. I usually only had a single trial anyways and wanted to load the last checkpoint as easily as possible (few required arguments).

I posted some related questions in the issue as well since someone was just asking about it: https://github.com/ray-project/ray/issues/7927#issuecomment-809573299 
I'm happy to change the function to whatever we need/want. Or to implement two (or three) wrapper functions if that helps.

Reply:
@krfricke I just updated the function to optionally accept a metric and mode like you suggested and then load the best trial according to these.
By making metric and mode optional, it's still possible to simply get the last checkpoint of a single trial without having to specify any arguments.

The function itself is now just a very simple wrapper of `get_best_checkpoint()`, but I think it's value is in being a simple, straight-forward way of getting the last checkpoint for users that are not familiar with all functions/API.

What do you think?

Reply:
Great! Just one last thing - I think the call in line 301 should always use `metric=training_iteration` and `mode=max`, right? Because then it will always fetch the last checkpoint, but select the best trial according to the passed metric and mode.

Reply:
(I assume this is what we want since the function is called `get_last_checkpoint`)

Reply:
Oh, of course. Thanks for noticing!